src/controllers/controllermanager.cpp

```diff
@@ -174,7 +178,20 @@ void ControllerManager::updateControllerList() {
 
     locker.relock();
     if (newDeviceList != m_controllers) {
+        // Check for keyboard controller
+        KeyboardController* pKeyboardController;
+        foreach (Controller* pController, newDeviceList) {
+            pKeyboardController = qobject_cast<KeyboardController*>(pController);
```

Comment:
If possible you should avoid up-casting. 
It looks like this loop can be done moved to the enumerators loop above. 
(It will also make the look regions smaller) 

It looks like you can remove KeyboardEnumerator() from m_enumerators and handle it separate. 
Or more object orientated: move the connecting to KeyboardController and add a new virtual empty implementation to the base class.